dex/src/main/scala/com/wavesplatform/dex/db/AssetsStorage.scala

```diff
@@ -0,0 +1,54 @@
+package com.wavesplatform.dex.db
+
+import java.util.concurrent.ConcurrentHashMap
+
+import com.wavesplatform.database.DBExt
+import com.wavesplatform.dex.MatcherKeys
+import com.wavesplatform.dex.effect.{FutureResult, liftValueAsync}
+import com.wavesplatform.dex.grpc.integration.dto.BriefAssetDescription
+import com.wavesplatform.transaction.Asset
+import com.wavesplatform.transaction.Asset.IssuedAsset
+import com.wavesplatform.transaction.assets.exchange.AssetPair
+import org.iq80.leveldb.DB
+
+trait AssetsStorage {
+  def put(asset: IssuedAsset, item: BriefAssetDescription): Unit
+  def get(asset: IssuedAsset): Option[BriefAssetDescription]
+}
+
+object AssetsStorage {
+
+  val wavesLifted: FutureResult[BriefAssetDescription] = liftValueAsync { BriefAssetDescription.wavesDescription }
+
+  def cache(db: DB): AssetsStorage = new AssetsStorage {
```

Comment:
Ideologically the cache is a decorator to some interface. 

```scala
def cache(db: DB)
```
doesn't compose well with possible backends (leveldb for this case) and depends on implementation details.

It should be:
```scala
def cache(inner: AssetsStorage): AssetsStorage
```